extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeTrendingExtractor.java

```diff
@@ -92,25 +91,64 @@ public String getName() throws ParsingException {
 
     @Nonnull
     @Override
-    public InfoItemsPage<StreamInfoItem> getInitialPage() {
+    public InfoItemsPage<StreamInfoItem> getInitialPage() throws ParsingException {
         final StreamInfoItemsCollector collector = new StreamInfoItemsCollector(getServiceId());
         final TimeAgoParser timeAgoParser = getTimeAgoParser();
-        final JsonArray itemSectionRenderers = initialData.getObject("contents")
-                .getObject("twoColumnBrowseResultsRenderer").getArray("tabs").getObject(0)
-                .getObject("tabRenderer").getObject("content").getObject("sectionListRenderer")
-                .getArray("contents");
-
-        for (final Object itemSectionRenderer : itemSectionRenderers) {
-            final JsonObject expandedShelfContentsRenderer = ((JsonObject) itemSectionRenderer)
-                    .getObject("itemSectionRenderer").getArray("contents").getObject(0)
-                    .getObject("shelfRenderer").getObject("content")
-                    .getObject("expandedShelfContentsRenderer");
-            for (final Object ul : expandedShelfContentsRenderer.getArray("items")) {
-                final JsonObject videoInfo = ((JsonObject) ul).getObject("videoRenderer");
-                collector.commit(new YoutubeStreamInfoItemExtractor(videoInfo, timeAgoParser));
-            }
+        final JsonObject tabContent = getTrendingTabRenderer().getObject("content");
```

Comment:
Maybe move the `.getObject("content")` to `getTrendingTabRenderer()`, adding `has("content")` to the filtering there?